Arduino/Propulsion/src/PropulsionModule.cpp

```diff
@@ -35,7 +41,21 @@ uint16_t byteToShort(int16_t index);
 void errorReply(uint8_t errorByte);
 
 /*****Propulsion Module Specific Parameters**************************/
+#ifdef FEATHER
+Adafruit_MotorShield motorShield = Adafruit_MotorShield();
```

Comment:
Is this a function of type *name* called *name* ?

Reply:
Sorry, I don't understand the question

Reply:
I see two objects called 'Adafruit_MotorShield'. Is one a keyword/type and one the name of the function? If so, should the function be named something different?

Reply:
This is one (of many) ways of constructing a new object (an instance of the `Adafruit_MotorShield` class). That is to say that here, `Adafruit_MotorShield` is both the type of the object (on the left-most bit of the left-hand side) and an invocation of the constructor (the right-hand side).

Reply:
I think it's also legally written as:

```diff
@@ -42,11 +42,11 @@ void errorReply(uint8_t errorByte);
 
 /*****Propulsion Module Specific Parameters**************************/
 #ifdef FEATHER
-Adafruit_MotorShield motorShield = Adafruit_MotorShield();
-Adafruit_DCMotor *m0 = motorShield.getMotor(3);
-Adafruit_DCMotor *m1 = motorShield.getMotor(4);
+Adafruit_MotorShield *motorShield = new Adafruit_MotorShield();
+Adafruit_DCMotor *m0 = motorShield->getMotor(3);
+Adafruit_DCMotor *m1 = motorShield->getMotor(4);
 void initDriver() {
-  motorShield.begin();
+  motorShield->begin();
   m0->run(RELEASE);
   m1->run(RELEASE);
 }
```

Reply:
Or if we want I think it can also be written like so:

```diff
@@ -42,7 +42,7 @@ void errorReply(uint8_t errorByte);
 
 /*****Propulsion Module Specific Parameters**************************/
 #ifdef FEATHER
-Adafruit_MotorShield motorShield = Adafruit_MotorShield();
+auto motorShield = Adafruit_MotorShield();
 Adafruit_DCMotor *m0 = motorShield.getMotor(3);
 Adafruit_DCMotor *m1 = motorShield.getMotor(4);
 void initDriver() {
```

Reply:
How does the auto work? Does it infer they type from the invocation of the constructor?

Reply:
Auto infers type. It's pretty neat and is a new feature in newer versions of C++. Read more [here](http://www.learncpp.com/cpp-tutorial/4-8-the-auto-keyword/)

Reply:
Yup, as @FifoIronton noted. I can switch to that if we think that's clearer.

Reply:
I'm cool with either, I was just confused at first.

Reply:
Sure, I'll update it to use `auto`